### PR TITLE
Add a special bundler binstub that ensures the correct version is activated

### PR DESCRIPTION
**Pull request by segiddins**
### What was the end-user problem that led to this PR?

Closes https://github.com/bundler/bundler/issues/5876.

> Once Bundler 2 ships, each application will depend on a 1.x or 2.x version of Bundler.

We want to make sure that the user always has an _easy_ and _convenient_ way to activate the version of Bundler their app is depending upon.

### What was your diagnosis of the problem?

> One way to ensure that the application will always get the version of Bundler that it needs is with an application-and-bundler-version specific binstub. Let's make it so that bundle binstubs bundler will generate a binstub for the correct version of Bundler, and simply running bin/bundle will always provide the correct version of Bundler.

### What is your fix for the problem, implemented in this PR?

My fix implements a `bundle` binstub when `bundle binstubs bundler` is run

### Why did you choose this fix out of the possible options?

I chose this fix because it allows us to dynamically select the correct Bundler version (with appropriate overrides, such as `$BUNDLER_VERSION` and `bundle update --bundler`), both when invoking `bin/bundle`, and also when invoking other bundler-generated binstubs.

### CODE REVIEW

File: lib/bundler/templates/Executable.bundler [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    return gemfile if gemfile && !gemfile.empty?
+
+    File.expand_path("../<%= relative_gemfile_path %>", __FILE__)
+  end
+
+  def lockfile
+    File.expand_path case File.basename(gemfile)
+                      when 'gems.rb' then gemfile.sub(/\.rb$/, gemfile)
+                      else "#{gemfile}.lock"
+                      end
```
Comment (indirect) [inline lib/bundler/templates/Executable.bundler]: Can you use a variable so the case statement body isn't indented?

Comment (segiddins) [inline lib/bundler/templates/Executable.bundler]: Sure, this was lifted directly from postit 👍🏻

**Comment by segiddins** [comment]
I actually have something different in mind for RubyGems, which will handle requiring Bundler/setup as well

**Comment by segiddins** [comment]
@indirect I believe this should work now, and I've made sure to run the file through rubocop

**Comment by segiddins** [comment]
Wow, green within a day! That's like a new record for me

**Comment by segiddins** [comment]
@indirect r?

**Comment by indirect** [comment]
Awesome. @bundlerbot r+

**Pull request merged**